src/dash-playlist-loader.js

```diff
@@ -65,6 +70,103 @@ export const updateMaster = (oldMaster, newMaster) => {
   return update;
 };
 
+export const generateSidxKey = (sidxInfo) => {
+  // should be non-inclusive
+  const sidxByteRangeEnd =
+    sidxInfo.byterange.offset +
+    sidxInfo.byterange.length -
+    1;
+
+  return sidxInfo.uri + '-' +
+    sidxInfo.byterange.offset + '-' +
+    sidxByteRangeEnd;
+};
+
+const equivalentSidx = (a, b) => {
+  let equivalentMap = true;
+
+  if (a.map && b.map) {
+    equivalentMap = a.map && b.map &&
+      a.map.byterange.offset === b.map.byterange.offset &&
+      a.map.byterange.length === b.map.byterange.length;
+  }
+
+  return equivalentMap &&
+    a.uri === b.uri &&
+    a.byterange.offset === b.byterange.offset &&
+    a.byterange.length === b.byterange.length;
+};
+
+// exported for testing
+export const compareSidxEntry = (playlists, oldSidxMapping) => {
+  const newSidxMapping = {};
+
+  for (const uri in playlists) {
+    const playlist = playlists[uri];
+    const currentSidxInfo = playlist.sidx;
+
+    if (currentSidxInfo) {
+      const key = generateSidxKey(currentSidxInfo);
+
+      if (!oldSidxMapping[key]) {
+        break;
+      }
+
+      const savedSidxInfo = oldSidxMapping[key].sidxInfo;
+
+      if (equivalentSidx(savedSidxInfo, currentSidxInfo)) {
+        newSidxMapping[key] = oldSidxMapping[key];
+      }
+    }
+  }
+
+  return newSidxMapping;
+};
+
+// exported for testing
+export const filterSidxMapping = (masterXml, srcUrl, clientOffset, oldSidxMapping) => {
```

Comment:
I don't quite understand what this function is filtering. It seems to be merging a new sidx mapping with an old one?

Reply:
it's removing sidx mappings that have changed (either url or indexRange)

Reply:
Maybe rename it to `removeChangedMappings` for clarity?

Reply:
I think filter is fine there because it parallels Array#filter. However, a comment can be added to clarify.

Reply:
Updated the name a bit as well.